src/context/fetch.ts

```diff
@@ -26,6 +28,23 @@ const createFetchRequestParameters = (input: MockedRequest): RequestInit => {
     return requestParameters
   }
 
+  // When the user made their original request, the browser added a boundary e.g.
+  // "content-type": multipart/form-data; boundary=----WebKitFormBoundarygyGKnRF8C9LT0BhB"
+  // If we now reuse this string as a user-provided header, the form-data will break and no payload will be send
+  // https://community.cloudflare.com/t/cannot-seem-to-send-multipart-form-data/163491/2
+  if (
+    input.headers.get('content-type')?.includes('multipart/form-data') &&
+    typeof body === 'object'
+  ) {
+    input.headers.delete('content-type')
```

Comment:
What would happen if instead of removing the entire `Content-Type` header we:

1. Parsed it.
2. Removed only the `boundary` part?

I have concerns that removing the header means we're entrusting the browser to guess the user's intentions. This may work in most cases but I think it may fail in other. Specifying a `Content-Type` is a user's indention, so if we can preserve the header but remove any automatically appended boundaries that'd be the least invasive solution. 

What do you think?

Reply:
I intercepted the requests made by Chrome with [HTTP Toolkit](https://httptoolkit.tech/).

There are two different issues:
- Chrome dev tools has a visual bug. The payload is always included, but displayed in three different ways in the dev tools (see screenshots below). There is no difference between not sending the `Content-Type` header, sending the header without a `boundary` and sending the header with a `boundary` (at least in Chrome and Firefox). The data send over the network is always the same. **So we don't need need to delete or modify the header at all**. Btw: Chrome and Firefox are adding their own `boundary` value whatever the user does. 
- However we still need to create a new `FormData` in `fetch.ts`

After the fix the correct data will be send over the network, but developers using Chrome will not realise this if they rely only on Chrome's dev tools, but don't inspect the network traffic.

# Chrome
### Sending no header
<img width="1680" alt="Screenshot 2022-04-04 at 18 32 07" src="https://user-images.githubusercontent.com/35801656/161590465-d7c14ba4-c6d5-4edc-a800-c52238ae90d3.png">
<img width="1680" alt="Screenshot 2022-04-04 at 18 32 14" src="https://user-images.githubusercontent.com/35801656/161590512-421c035a-121e-4191-9b98-a8eca5e0e4d2.png">

### Sending `'content-type': 'multipart/form-data'`
<img width="1680" alt="Screenshot 2022-04-04 at 18 33 07" src="https://user-images.githubusercontent.com/35801656/161590596-2e479077-3633-4c76-98b2-609d343ff7fb.png">
<img width="1680" alt="Screenshot 2022-04-04 at 18 33 12" src="https://user-images.githubusercontent.com/35801656/161590611-4e3a83e8-23e2-4778-994e-1d3dec0e4e09.png">

### Sending ` 'content-type': 'multipart/form-data; boundary=----WebKitFormBoundarygyGKnRF8C9LT0BhB'`
<img width="1680" alt="Screenshot 2022-04-04 at 18 33 28" src="https://user-images.githubusercontent.com/35801656/161590724-930e6b5a-7256-4fc1-987c-a40c00484062.png">
<img width="1680" alt="Screenshot 2022-04-04 at 18 33 40" src="https://user-images.githubusercontent.com/35801656/161590748-18447c97-2675-4c21-a9d6-3699ea61f62a.png">

# HTTP Toolkit
### Before this PR
<img width="1680" alt="no-new-form-data" src="https://user-images.githubusercontent.com/35801656/161607283-9dbfa29a-3f41-44a9-b327-d528772bed0e.png">

### Creating new form data, but not modifying the header
<img width="1680" alt="new-form-data-only-sends-payload" src="https://user-images.githubusercontent.com/35801656/161607355-ae30266c-2161-4cb6-bb40-ec46811bda4a.png">

Reply:
Can I just take a moment to appreciate your effort on this? You've done a fantastic job getting to the core of the issue 👏 

I see now. It's a matter of re-constructing the `FormData`. I still think that's because the intercepter `req.body` will be text received from the worker/Node.js interceptor. I'm not comfortable to assess if we should extend upon the `IsomorphicRequest` to allow deriving textual request data to other formats similar to have `window.Request` does. This may be a good idea for the future discussions. 

For now, I agree with the point to address the issue at its immediate context: `ctx.fetch`.